### PR TITLE
Hw20 test

### PR DESCRIPTION
**Pull request by r0var**

### CODE REVIEW

File: HW20/group_vars/ubuntu/ubuntu.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ansible_user: vagrant
```
Comment (mavrik90) [inline HW20/group_vars/ubuntu/ubuntu.yaml]: Put those variable to another "general" group OR to all hosts group

File: HW20/hosts.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+all:
+  children:
```
Comment (mavrik90) [inline HW20/hosts.yaml:2]: to make a bit "smaller" try to define groups after you specify groups with hosts.
For example
```
all:
  children:
    testhost1:
      hosts:
        172.30.128.3
    testhost2:
      hosts:
        172.30.128.4
    centos:
      hosts:
        172.30.128.5
    mix:
      children:
        testhost1:
        centos:

File: HW20/roles/hw201/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    url: https://download.docker.com/linux/centos/docker-ce.repo
+    dest: /etc/yum.repos.d/docker-ce.repo
+  when: ansible_distribution == 'CentOS'
+
+- name: Install packages
+  become: true
+  package:
+    name: "{{item}}"
+    state: present
+  loop:
```
Comment (mavrik90) [inline HW20/roles/hw201/tasks/main.yaml]: set it as a variable

File: HW20/roles/hw201/tasks/main.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      apt_repository:
+        repo: deb https://download.docker.com/linux/ubuntu focal stable
+        state: present
+    - name: update apt cache
+      apt: update_cache=yes
+  when: ansible_distribution == 'Ubuntu'
+
+- name: Add Docker CE repo centos
+  become: true
+  get_url:
```
Comment (mavrik90) [inline HW20/roles/hw201/tasks/main.yaml:19]: Use loop here to be able to download more than one file

File: HW20/roles/hw201/tasks/sudoadd.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (mavrik90) [inline HW20/roles/hw201/tasks/sudoadd.yaml:1]: Replace with only one task. Use variables to do so.

File: HW20/roles/hw202/tasks/main.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
+---
+- name: copy local files to remote
+  copy:
+    src: "{{copysource}}"
+    dest: "{{copypath}}"
+    owner: vagrant
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/main.yaml]: use variables here

File: HW20/roles/hw202/tasks/main.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    group: vagrant
+    mode: '0644'
+
+- name: Install curl from deb
+  block:
+    - name: copy deb package
+      copy:
+        src: files/curl_7.68.0-1ubuntu2.5_amd64.deb
+        dest: /home/vagrant
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/main.yaml]: use variable here

File: HW20/roles/hw202/tasks/main.yaml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    - name: install deb package
+      become: true
+      apt:
+        deb: /home/vagrant/curl_7.68.0-1ubuntu2.5_amd64.deb
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/main.yaml]: variable instead of hardcoded path

Comment (mavrik90) [reply HW20/roles/hw202/tasks/main.yaml]: Loop it make it scalable

File: HW20/roles/hw202/tasks/main.yaml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        state: present
+  when: ansible_distribution == 'Ubuntu'
+
+- name: Install curl from rpm
+  block:
+    - name: copy rpm package
+      copy:
+        src: files/curl-7.61.1-14.el8_3.1.x86_64.rpm
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/main.yaml]: variables and loop

File: HW20/roles/hw202/tasks/main.yaml [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        dest: /home/vagrant
+    - name: install rpm package
+      become: true
+      yum:
+        name: /home/vagrant/curl-7.61.1-14.el8_3.1.x86_64.rpm
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/main.yaml]: variables and loop

File: HW20/roles/hw202/tasks/main.yaml [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+        state: present
+        disable_gpg_check: yes
+  when: ansible_distribution == 'CentOS'
+
+- name: create hw20 file info
+  become: true
+  template:
+    src: templates/hw20.j2
```
Comment (mavrik90) [inline HW20/roles/hw202/tasks/main.yaml]: loop here

File: HW20/group_vars/ubuntu/ubuntu.yaml (hunk 1 above)
Comment (r0var) [reply HW20/group_vars/ubuntu/ubuntu.yaml:1]: Sounds good, doesn't work.

File: HW20/hosts.yaml (hunk 2 above)
Comment (r0var) [reply HW20/hosts.yaml:2]: In my case it will give more lines.

File: HW20/roles/hw201/tasks/main.yaml (hunk 4 above)
Comment (r0var) [reply HW20/roles/hw201/tasks/main.yaml:19]: I'm not just downloading the file. I also save it to the repositories folder. It only works like that. I do not need to download anything else to the folder with repositories.  yum_repository corrupts the repository.

Comment (r0var) [reply HW20/roles/hw201/tasks/main.yaml:19]: It is not just downloading file. yum_repository corrupts docker repostory file. So i exclusively save docker repo file to repo folder. Maybe in other cases yum_repository will be work fine.

**Review by mavrik90 — APPROVED** [review]
LGTM

**Pull request merged**